WordPress/Classes/ViewRelated/Gutenberg/GutenbergMediaPickerHelper.swift

```diff
@@ -38,12 +38,36 @@ final class GutenbergMediaPickerHelper: NSObject {
         context.present(picker, animated: true)
     }
 
-    func presentSiteMediaPicker(filter: WPMediaType, allowMultipleSelection: Bool, completion: @escaping GutenbergMediaPickerHelperCallback) {
+    func presentSiteMediaPicker(filter: WPMediaType, allowMultipleSelection: Bool, initialSelection: [Int] = [], completion: @escaping GutenbergMediaPickerHelperCallback) {
         didPickMediaCallback = completion
-        MediaPickerMenu(viewController: context, filter: .init(filter), isMultipleSelectionEnabled: allowMultipleSelection)
+        let initialMediaSelection = mapMediaIdsToMedia(initialSelection)
+        MediaPickerMenu(viewController: context, filter: .init(filter), isMultipleSelectionEnabled: allowMultipleSelection, initialSelection: initialMediaSelection)
             .showSiteMediaPicker(blog: post.blog, delegate: self)
     }
 
+    private func mapMediaIdsToMedia(_ mediaIds: [Int]) -> [Media] {
```

Comment:
Any `Media` objects returned from this method aren't thread-safe – it looks like a lot of what's going on here _should_ happen on the main thread, but can we add assertions to ensure that is the case?

We could also convert these to thread-safe objects, but that might be overkill at this point?

Reply:
Thank you for noting this. I am quite unfamiliar with the subject of thread safety, so I appreciate that inquiries and suggestions. 

I added an assertion in 186749917facf611a2f4e78d012674c08409c30f based on what I understand of your feedback. Please let me know if I misunderstood or you have alternatives to suggest.